pkg/controller/expcertificates/issuing/issuing_controller.go

```diff
@@ -0,0 +1,396 @@
+	})
+
+	// build a list of InformerSynced functions that will be returned by the Register method.
+	// the controller will only begin processing items once all of these informers have synced.
+	mustSync := []cache.InformerSynced{
+		certificateRequestInformer.Informer().HasSynced,
+		secretsInformer.Informer().HasSynced,
+		certificateInformer.Informer().HasSynced,
+	}
+
+	return &controller{
+		certificateLister:        certificateInformer.Lister(),
+		certificateRequestLister: certificateRequestInformer.Lister(),
+		secretLister:             secretsInformer.Lister(),
+		client:                   client,
+		recorder:                 recorder,
+		clock:                    clock,
+
+		//enableSecretOwnerReferences: ctx.CertificateOptions.EnableOwnerRef
+	}, queue, mustSync
+}
+
+func (c *controller) ProcessItem(ctx context.Context, key string) error {
+	log := logf.FromContext(ctx).WithValues("key", key)
+	namespace, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		return nil
+	}
+
+	crt, err := c.certificateLister.Certificates(namespace).Get(name)
+	if apierrors.IsNotFound(err) {
+		log.Error(err, "certificate not found for key")
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+
+	if !apiutil.CertificateHasCondition(crt, cmapi.CertificateCondition{
+		Type:   cmapi.CertificateConditionIssuing,
+		Status: cmmeta.ConditionTrue,
+	}) {
+		// Do nothing if an issuance is not in progress.
+		return nil
+	}
+
+	if crt.Status.NextPrivateKeySecretName == nil ||
+		len(*crt.Status.NextPrivateKeySecretName) == 0 {
+		// Do nothing if the next private key secret name is not set
+		return nil
+	}
+
+	nextPrivateKeySecretName := *crt.Status.NextPrivateKeySecretName
+
+	reqs, nextRevision, err := findNextRevisionCertificateRequestsForCertificate(log, crt, c.certificateRequestLister)
+	if err != nil {
+		return err
+	}
+
+	if len(reqs) != 1 {
+		// If none exist do nothing. If multiple exist, then leave to
+		// requestmanager controller to clean up.
+		return nil
+	}
+
+	req := reqs[0]
+
+	reqReason := apiutil.CertificateRequestReadyReason(req)
+	switch reqReason {
+
+	// If the certificate request has failed, set the last failure time to now,
+	// and set the Issuing status condition to False with reason.
+	case cmapi.CertificateRequestReasonFailed:
+		nowTime := metav1.NewTime(c.clock.Now())
+		crt.Status.LastFailureTime = &nowTime
+
+		var reason, message string
+
+		reason = "FailedToIssue"
+		message = fmt.Sprintf("The certificate request has failed to issue and will be retired: %s",
+			apiutil.CertificateRequestFailedMessage(req))
+
+		crt = crt.DeepCopy()
+		apiutil.SetCertificateCondition(crt, cmapi.CertificateConditionIssuing, cmmeta.ConditionFalse, reason, message)
+
+		if err := c.updateCertificateStatus(ctx, crt); err != nil {
+			return err
+		}
+
+		c.recorder.Event(crt, corev1.EventTypeWarning, "Issuing", message)
```

Comment:
`Issuing` feels like the wrong 'reason' field for the event here... perhaps `Failed`? Or otherwise copy it from the CertificateRequest